src/Security/RolesMatrixBuilder.php

```diff
@@ -0,0 +1,335 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\UserBundle\Security;
+
+use Sonata\AdminBundle\Admin\Pool;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+use Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface;
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * @author Silas Joisten <silasjoisten@hotmail.de>
+ */
+class RolesMatrixBuilder
```

Comment:
should this one also be `final`?

Reply:
If this is final i dont know how to fix the warning in test:

```console
1) Sonata\UserBundle\Tests\Form\Type\RolesMatrixTypeTest::testGetDefaultOptions
Class "Sonata\UserBundle\Security\RolesMatrixBuilder" is declared "final" and cannot be mocked.

2) Sonata\UserBundle\Tests\Form\Type\RolesMatrixTypeTest::testGetParent
Class "Sonata\UserBundle\Security\RolesMatrixBuilder" is declared "final" and cannot be mocked.

3) Sonata\UserBundle\Tests\Form\Type\RolesMatrixTypeTest::testSubmitValidData
Class "Sonata\UserBundle\Security\RolesMatrixBuilder" is declared "final" and cannot be mocked.

4) Sonata\UserBundle\Tests\Form\Type\RolesMatrixTypeTest::testSubmitInvalidData
Class "Sonata\UserBundle\Security\RolesMatrixBuilder" is declared "final" and cannot be mocked.

5) Sonata\UserBundle\Tests\Form\Type\RolesMatrixTypeTest::testChoicesAsValues
Class "Sonata\UserBundle\Security\RolesMatrixBuilder" is declared "final" and cannot be mocked.

6) Sonata\UserBundle\Tests\Twig\RolesMatrixExtensionTest::testGetName
Class "Sonata\UserBundle\Security\RolesMatrixBuilder" is declared "final" and cannot be mocked.

7) Sonata\UserBundle\Tests\Twig\RolesMatrixExtensionTest::renderCustomRolesList
Class "Sonata\UserBundle\Security\RolesMatrixBuilder" is declared "final" and cannot be mocked.

8) Sonata\UserBundle\Tests\Twig\RolesMatrixExtensionTest::renderMatrix
Class "Sonata\UserBundle\Security\RolesMatrixBuilder" is declared "final" and cannot be mocked.
```

Reply:
Why dont you create an interface for this class and then you can mock the interface instead of the direct implementation?

Reply:
If this is the way we should go, fine.
And use the interface for typehinting instead of the class name? 

Which name do you prefer for the interface?